src/Common/PipeWriterStream.cs

```diff
@@ -83,7 +83,16 @@ private ValueTask WriteCoreAsync(ReadOnlyMemory<byte> source, CancellationToken
 
             return default;
 
-            async ValueTask WriteSlowAsync(ValueTask<FlushResult> flushTask) => await flushTask;
+            async ValueTask WriteSlowAsync(ValueTask<FlushResult> flushTask)
+            {
+                var flushResult = await flushTask;
+
+                // Cancellation can be triggered by PipeWriter.CancelPendingFlush
+                if (flushResult.IsCanceled)
```

Comment:
Are we handling this on the client properly?

https://github.com/aspnet/SignalR/blob/f632330d7f5232053ef183c8c78b7c367087ff37/src/Microsoft.AspNetCore.Http.Connections.Client/Internal/PipeReaderFactory.cs#L33

https://github.com/aspnet/SignalR/blob/9d8b83e6b34f294cc5aba4f94fdd74cb8e5bc0d7/src/Microsoft.AspNetCore.Http.Connections.Client/Internal/LongPollingTransport.cs#L187

Reply:
https://github.com/aspnet/SignalR/blob/f632330d7f5232053ef183c8c78b7c367087ff37/src/Microsoft.AspNetCore.Http.Connections.Client/Internal/PipeReaderFactory.cs#L33

Already handles OperationCanceledException.

https://github.com/aspnet/SignalR/blob/9d8b83e6b34f294cc5aba4f94fdd74cb8e5bc0d7/src/Microsoft.AspNetCore.Http.Connections.Client/Internal/LongPollingTransport.cs#L187

Hmmm... the FlushAsync after CopyToAsync seems odd. CopyToAsync already flushes, no?

This is nested inside OperationCanceledException catch so is functionally similar to breaking out of the loop